src/modules/pythonmonkey/pythonmonkey.cc

```diff
@@ -48,10 +48,25 @@
 
 JS::PersistentRootedObject jsFunctionRegistry;
 
-void finalizationRegistryGCCallback(JSContext *cx, JSGCStatus status, JS::GCReason reason, void *data) {
+void pythonmonkeyGCCallback(JSContext *cx, JSGCStatus status, JS::GCReason reason, void *data) {
   if (status == JSGCStatus::JSGC_END) {
     JS::ClearKeptObjects(GLOBAL_CX);
     while (JOB_QUEUE->runFinalizationRegistryCallbacks(GLOBAL_CX));
+
+    if (_Py_IsFinalizing()) {
+      return; // do not move char pointers around if python is finalizing
+    }
+
+    JS::AutoCheckCannotGC nogc;
+    for (const JSStringProxy *jsStringProxy: jsStringProxies) { // char buffers may have moved, so we need to re-point our JSStringProxies
```

Comment:
In symmetry to this, would a Python str object move its char buffers, and how to notify the JS land?

Reply:
Do you know if python string objects move their buffers? As far as I'm aware they never move